Added Pangolin Connector

prodesert22 opened the pull request:
Pangolin is a Dex on Avalanche, similar to Uniswap

- [x] Review and approve connector @pradyuman-verma 
- [x] Deploy the contract and comment the address
- [x] Created resolver. found [here](https://github.com/Instadapp/dsa-resolvers/pull/42)
- [ ] Add to dsa-connect

prodesert22 committed:
Added Pangolin Exchange Connector

Fixed Contract

Add: test to Pangolin connector

Fix InstaDapp Addresses/Scripts/PNG Connector

Removed unnecessary scripts

Moved PNG connector contract and Connector tests

pradyuman-verma commented on scripts/getMasterSigner_avalanche.js:
```diff
@@ -0,0 +1,24 @@
+  });
+
+  return await ethers.getSigner(masterAddress);
+};
```
This script is not needed now, as we have updated the original script.

pradyuman-verma commented on scripts/buildDSAv2_avalanche.js:
```diff
@@ -0,0 +1,14 @@
+    const receipt = await tx.wait()
+    const event = receipt.events.find(a => a.event === "LogAccountCreated")
+    return await ethers.getContractAt(instaImplementations_m1.abi, event.args.account)
+};
```
This is not needed too.

prodesert22 committed:
Add Pangolin Stake Connector and Tests

Resolve errors

Resolve review issues

Fix comment

Co-authored-by: 0xPradyuman <63545809+pradyuman-verma@users.noreply.github.com>

Added NatSpec to Staking Connector

Fix NatSpecs

pradyuman-verma commented on test/pangolin/pangolin.test.js:
```diff
@@ -0,0 +1,204 @@
+const encodeSpells = require("../../scripts/encodeSpells.js")
+const getMasterSigner = require("../../scripts/getMasterSigner_avalanche")
+
+const connectV2PangolinArtifacts = require("../../artifacts/contracts/avalanche/connectors/pangolin/main.sol/ConnectV2PngAvalanche.json");
```
```suggestion
import { ConnectV2PngAvalanche__factory } from "../../../typechain";
```

prodesert22 replied:
This file was moved to pangolin_exchange.test.ts, please ignore this file, It was a mistake of mine not erased it

pradyuman-verma commented on test/pangolin/pangolin.test.js:
```diff
@@ -0,0 +1,204 @@
+        // Deploy and enable Pangolin Connector
+        pangolinConnector = await deployAndEnableConnector({
+            connectorName: pangolinConnectorName,
+            contractArtifact: connectV2PangolinArtifacts,
```
```suggestion
            contractArtifact: ConnectV2PngAvalanche__factory,
```

pradyuman-verma commented on test/avalanche/pangolin/pangolin_exchange.test.ts:
```diff
@@ -0,0 +1,283 @@
+    let dsaWallet0: any;
+    let masterSigner: any;
+    let instaConnectorsV2: any;
+    let pangolinConnector: any;
```
```suggestion
    let dsaWallet0: Signer;
    let masterSigner: Signer;
    let instaConnectorsV2: Contract;
    let pangolinConnector: Contract;
```

prodesert22 replied:
I tested and show an error message saying that dsaWallet0 is a contract and not a signer

pradyuman-verma replied:
Oh yes, it is indeed a contract.

pradyuman-verma commented on test/avalanche/pangolin/pangolin_exchange.test.ts:
```diff
@@ -0,0 +1,283 @@
+    it("Should have contracts deployed.", async function () {
+        expect(!!instaConnectorsV2.address).to.be.true;
+        expect(!!pangolinConnector.address).to.be.true;
+        expect(!!masterSigner.address).to.be.true;
```
```suggestion
        expect(!!(await masterSigner.getAddress())).to.be.true;
```

pradyuman-verma commented on test/avalanche/pangolin/pangolin_exchange.test.ts:
```diff
@@ -0,0 +1,283 @@
+    
+    describe("DSA wallet setup", function () {
+        it("Should build DSA v2", async function () {
+            dsaWallet0 = await buildDSAv2(wallet0.address)
```
Replace everyplace where it is used!
```suggestion
            dsaWallet0 = await buildDSAv2(wallet0.getAddress())
```

pradyuman-verma commented on test/pangolin/pangolin.test.js:
```diff
@@ -0,0 +1,204 @@
+const hre = require("hardhat");
+
+const { web3, deployments, waffle, ethers } = hre;
+const { provider, deployContract } = waffle
```
```suggestion

```

pradyuman-verma commented on test/pangolin/pangolin.test.js:
```diff
@@ -0,0 +1,204 @@
+    let pangolinConnector;
+    
+    const wallets = provider.getWallets()
+    const [wallet0, wallet1, wallet2, wallet3] = wallets
```
```suggestion
[wallet0, wallet1] = await ethers.getSigners();
```

pradyuman-verma commented on test/pangolin/pangolin.test.js:
```diff
@@ -0,0 +1,204 @@
+    let dsaWallet0
+    let masterSigner;
+    let instaConnectorsV2;
+    let pangolinConnector;
```
```suggestion
    let pangolinConnector: Contract;
  let wallet0: Signer, wallet1:Signer;
  let dsaWallet0: any;
  let instaConnectorsV2: Contract;
  let masterSigner: Signer;
```

pradyuman-verma commented on test/pangolin/pangolin.test.js:
```diff
@@ -0,0 +1,204 @@
+    it("Should have contracts deployed.", async function () {
+        expect(!!instaConnectorsV2.address).to.be.true;
+        expect(!!pangolinConnector.address).to.be.true;
+        expect(!!masterSigner.address).to.be.true;
```
```suggestion
 expect(!!(await masterSigner.getAddress())).to.be.true;
```

pradyuman-verma commented on test/pangolin/pangolin.test.js:
```diff
@@ -0,0 +1,204 @@
+    
+    describe("DSA wallet setup", function () {
+        it("Should build DSA v2", async function () {
+            dsaWallet0 = await buildDSAv2(wallet0.address)
```
```suggestion
           dsaWallet0 = await buildDSAv2(wallet0.getAddress());
```

pradyuman-verma replied:
replace at other places too

pradyuman-verma commented on test/pangolin/pangolin.test.js:
```diff
@@ -0,0 +1,204 @@
+const { expect } = require("chai");
```
move `test/pangolin/pangolin.test.js` to `test/avalanche/pangolin/pangolin.test.js`

pradyuman-verma commented on test/pangolin/pangolin.test.js:
```diff
@@ -0,0 +1,204 @@
+const buildDSAv2 = require("../../scripts/buildDSAv2_avalanche")
+const deployAndEnableConnector = require("../../scripts/deployAndEnableConnector.js")
+const encodeSpells = require("../../scripts/encodeSpells.js")
+const getMasterSigner = require("../../scripts/getMasterSigner_avalanche")
```
update them as well.
Some things here are not needed as the repo has been updated to support multichain support !

pradyuman-verma commented on contracts/avalanche/connectors/pangolin/exchange/main.sol:
```diff
@@ -0,0 +1,196 @@
+     * @param tokenA The address of token A.(For AVAX: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+     * @param tokenB The address of token B.(For AVAX: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+     * @param amtA The amount of A tokens to deposit.
+     * @param unitAmt The unit amount of of amtB/amtA with slippage.
```
```suggestion
     * @param unitAmt The unit amount of amtB/amtA with slippage.
```
